language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/util/FieldAccessCompletionResolver.java

```diff
@@ -287,6 +288,10 @@ private Symbol getSymbolByName(List<Symbol> visibleSymbols, String name, @Nonnul
                         .collect(Collectors.toList());
                 visibleEntries.addAll(methodSymbols);
                 break;
+            case INTERSECTION:
+                TypeSymbol effectiveTypeDesc = ((IntersectionTypeSymbol) rawType).effectiveTypeDescriptor();
+                visibleEntries.addAll(getVisibleEntries(effectiveTypeDesc, node));
+                break;
```

Comment:
Why do we have this here instead of within the `CommonUtil.getRawType`? Since this is a common requirement for all other places, shall we move it there?

Reply:
Done